File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/06.Nested-Loop/02-exercise/02.equalSumsEvenOddPosition.js
function equalSumsEvenOddPosition(input) {

    const firstNum = Number(input[0]);
    const secondNum = Number(input[1]);

    let printLine = "";

    for (let i = firstNum; i <= secondNum; i++) {
        let currentNum = i + "";
        let evenSum = 0;
        let oddSum = 0;
        for (let j = 0; j <= currentNum.length; j++) {
            let currentDigit = Number(currentNum.charAt(j));
            if (j % 2 === 0) {
                evenSum += currentDigit;
            } else {
                oddSum += currentDigit;
            }
        }

        if (evenSum === oddSum) {
            printLine += currentNum + " ";
        }
    }
    console.log(printLine);
}
equalSumsEvenOddPosition(["100000", "100050"]);

File: Python_Basics/Homeworks-And-Labs/06.nested_loops/02.exercise/02.equal_sums_even_odd_position.py
first_num = int(input())
second_num = int(input())

print_line = ""

for i in range(first_num, second_num + 1):
    current_num = str(i)
    even_sum = 0
    odd_sum = 0
    for j in range(len(current_num)):
        current_digit = int(current_num[j])
        if j % 2 == 0:
            even_sum += current_digit
        else:
            odd_sum += current_digit
            
    if even_sum == odd_sum:
        print_line += str(current_num) + " "
        
print(print_line)
